lib/open_food_network/address_finder.rb

```diff
@@ -79,7 +79,7 @@ def last_used_bill_address
 
     def last_used_ship_address
       return nil unless allow_search_by_email?
-      Spree::Order.complete.joins(:ship_address, :shipping_method).order('id DESC')
+      Spree::Order.complete.joins(:ship_address, shipments: :shipping_methods).order('id DESC')
```

Comment:
Did this join between shipments and shipping_methods show up somewhere else already? In that case it might be the right time to encapsulate it in an AR scope. This way we could refer back to it but in a decoupled way.

Reply:
yes, it is coming up here as well: https://github.com/openfoodfoundation/openfoodnetwork/pull/2680/commits/75656668cdaade2bbc2a994f1b38642b73734b03

it's only 2 occurrences and they are different because one joins to ship_methods and the other joins to ship_methods and ship_address. this is a reason no to do it.

but the real reason I didnt do it it's because I dont know how to do it. can you be a bit more specific about what would you like to see here? sorry for my AR ignorance. thanks!

Reply:
is this a new line on order_decorator?
has_many :shipping_methods, through: :shipments

Reply:
In that case it may not be worth doing anything but FYI https://guides.rubyonrails.org/v3.2/active_record_querying.html#scopes

Reply:
it would be
scope :with_shipping_methods, lambda {
    joins( ‘shipments: :shipping_methods’ )
end

ah, the merge one is nice.
thanks a lot for the links!!
ok, we keep it as it is. we will do this on a third occurrence for sure.